resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/SslConfigurationProvider.scala

```diff
@@ -112,92 +134,197 @@ private[spark] class SslConfigurationProvider(
     resolvedKeyStore.foreach {
-      resolvedSparkConf.set(KUBERNETES_DRIVER_SUBMIT_KEYSTORE, _)
+      resolvedSparkConf.set(DRIVER_SUBMIT_SSL_KEYSTORE, _)
     }
-    sparkConf.get(KUBERNETES_DRIVER_SUBMIT_TRUSTSTORE).foreach { trustStore =>
-      val trustStoreURI = Utils.resolveURI(trustStore)
-      trustStoreURI.getScheme match {
-        case "file" | null =>
-          resolvedSparkConf.set(KUBERNETES_DRIVER_SUBMIT_TRUSTSTORE, trustStoreURI.getPath)
-        case _ => throw new SparkException(s"Invalid trustStore URI $trustStore; trustStore URI" +
-          " for submit server must have no scheme, or scheme file://")
-      }
+    val (isLocalServerCertPem, resolvedServerCertPem): (Boolean, Option[String]) =
+      resolveLocalFile(maybeServerCertPem, "server cert PEM")
+    val (isLocalKeyPem, resolvedKeyPem) = resolveLocalFile(maybeKeyPem, "key PEM")
+    maybeTrustStore.foreach { trustStore =>
+      require(KubernetesFileUtils.isUriLocalFile(trustStore), s"Invalid trustStore URI" +
+        s"$trustStore; trustStore URI for submit server must have no scheme, or scheme file://")
+      resolvedSparkConf.set(DRIVER_SUBMIT_SSL_TRUSTSTORE, Utils.resolveURI(trustStore).getPath)
+    }
+    val clientCertPem: Option[String] = maybeClientCertPem.map { clientCert =>
+      require(KubernetesFileUtils.isUriLocalFile(clientCert), "Invalid client certificate PEM URI" +
+        s" $clientCert: client certificate URI must have no scheme, or scheme file://")
+      Utils.resolveURI(clientCert).getPath
     }
     val securityManager = new SparkSecurityManager(resolvedSparkConf)
-    (securityManager.getSSLOptions(KUBERNETES_SUBMIT_SSL_NAMESPACE), isLocalKeyStore)
+    val storeBasedSslOptions = securityManager.getSSLOptions(DRIVER_SUBMIT_SSL_NAMESPACE)
+    SubmissionSSLOptions(
+      storeBasedSslOptions,
+      isLocalKeyStore,
+      resolvedKeyPem.map(new File(_)),
+      isLocalKeyPem,
+      resolvedServerCertPem.map(new File(_)),
+      isLocalServerCertPem,
+      clientCertPem.map(new File(_)))
+  }
+
+  private def resolveLocalFile(file: Option[String],
+      fileType: String): (Boolean, Option[String]) = {
+    file.map { f =>
+      require(isValidSslFileScheme(f), s"Invalid $fileType URI $f, $fileType URI" +
+        s" for submit server must have scheme file:// or local:// (no scheme defaults to file://")
+      val isLocal = KubernetesFileUtils.isUriLocalFile(f)
+      (isLocal, Option.apply(Utils.resolveURI(f).getPath))
+    }.getOrElse(false, None)
+  }
+
+  private def validatePemsDoNotConflictWithStores(
+      maybeKeyStore: Option[String],
+      maybeTrustStore: Option[String],
+      maybeKeyPem: Option[String],
+      maybeServerCertPem: Option[String],
+      maybeClientCertPem: Option[String]) = {
+    maybeKeyPem.orElse(maybeServerCertPem).foreach { _ =>
+      require(maybeKeyStore.isEmpty,
+        "Cannot specify server PEM files and key store files; must specify only one or the other.")
```

Comment:
`jks` isn't the only type of KeyStore Java supports; e.g. we can also specify `pkcs12` and `bks`.